test/utils/formatAmount.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { ZERO, ONE, ALLOWANCE_VALUE } from 'const'
+import { formatAmount, toWei } from 'utils'
+import BN from 'bn.js'
+
+describe('Integer amounts', () => {
+  test('1 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1'), 'ether')))).toEqual('1')
+  })
+
+  test('12,345 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('12345'), 'ether')))).toEqual('12,345')
+  })
+
+  test('0100 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('0100'), 'ether')))).toEqual('100')
+  })
+
+  test('123,456,789,012 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('123456789012'), 'ether')))).toEqual('123,456,789,012')
+  })
+})
+
+describe('Exact decimal amounts', () => {
+  test('0.5 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('500'), 'milliether')))).toEqual('0.5')
+  })
+
+  test('1.234 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234'), 'milliether')))).toEqual('1.234')
+  })
+
+  test('1.2345 Ether', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234500'), 'microether')))).toEqual('1.2345')
+  })
+})
+
+describe('Rounding amounts', () => {
+  test('0', async () => {
+    expect(formatAmount(ZERO)).toEqual('0')
+  })
+
+  test('1 wei', async () => {
+    expect(formatAmount(ONE)).toEqual('0')
+  })
+
+  test('1000 wei', async () => {
+    expect(formatAmount(new BN('1000'))).toEqual('0')
+  })
+
+  test('1.23451 Ether, round down', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234510'), 'microether')))).toEqual('1.2345')
+  })
+
+  test('1.23456 Ether, round up', async () => {
+    expect(formatAmount(new BN(toWei(new BN('1234560'), 'microether')))).toEqual('1.2346')
+  })
+
+  test('1.234567890123456789 Ether', async () => {
+    expect(formatAmount(new BN('1234567890123456789'))).toEqual('1.2346')
+  })
+})
+
+describe('Tokens with precision 6', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatAmount(amount, 6)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('12345000000')
+    expect(formatAmount(amount, 6)).toEqual('12,345')
+  })
+
+  test('4,567,890.123444 units, round down', async () => {
+    const amount = new BN('4567890123444')
+    expect(formatAmount(amount, 6)).toEqual('4,567,890.1234')
+  })
+
+  test('4,567,890.123455 units, round up', async () => {
+    const amount = new BN('4567890123455')
+    expect(formatAmount(amount, 6)).toEqual('4,567,890.1235')
+  })
+})
+
+describe('2 decimals', () => {
+  test('1 unit', async () => {
+    const amount = new BN('1000000')
+    expect(formatAmount(amount, 6, 2)).toEqual('1')
+  })
+
+  test('12,345 units', async () => {
+    const amount = new BN('1000000')
+    expect(formatAmount(amount, 6, 2)).toEqual('1')
```

Comment:
The amount doesn't match the test name: `12,345` vs `1000000`

Reply:
fixed